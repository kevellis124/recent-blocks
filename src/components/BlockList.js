import React from "react";
import PropTypes from "prop-types";
import { Button } from "react-bootstrap";
import Block from "./Block";
function BlockList(props) {
  return (
    <div id="BlockList">
      <Button size={"lg"} onClick={props.reload}>
        Load
      </Button>
      {props.blocks.map(block => {
        return (
          <Block
            key={block.id}
            id={block.id}
            timestamp={block.timestamp}
            actionCount={block.transactions.length}
          />
        );
      })}
    </div>
  );
}

BlockList.propTypes = {
  blocks: PropTypes.array,
  reload: PropTypes.func
};

BlockList.defaultProps = {
  blocks: []
};

export default BlockList;
