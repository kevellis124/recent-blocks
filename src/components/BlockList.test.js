import { shallow } from '../enzyme';
import * as React from 'react';
import BlockList from './BlockList';

describe('<BlockList />', () => {
    describe('By default', () => {
        let component;
        beforeAll(() => {
            component = shallow(<BlockList/>);
        });
        it('hasnt changed', () => {
            expect(component.html()).toMatchSnapshot();
        });
    });
});