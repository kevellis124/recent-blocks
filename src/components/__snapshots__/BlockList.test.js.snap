// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockList /> By default hasnt changed 1`] = `"<div id=\\"BlockList\\"><button type=\\"button\\" class=\\"btn btn-primary btn-lg\\">Load</button></div>"`;
